MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -216,67 +217,234 @@ namespace WPEFramework {
+                    cmd += " &";
+                    cmd += "\0";
+                    m_task_map[tasks[i]]=true;
+                    if ( !m_abort_flag ){
+                        LOGINFO("Starting Script (SM) :  %s \n",cmd.c_str());
+                        system(cmd.c_str());
                     }
+                }
             }
-
             m_abort_flag=false;
             LOGINFO("Worker Thread Completed");
-            if (false == internetConnectStatus) {
+            if ( false == internetConnectStatus ) {
                 MaintenanceManager::_instance->onMaintenanceStatusChange(MAINTENANCE_ERROR);
                 LOGINFO("Maintenance completed as it is offline mode");
             }
         }
 
+        const string MaintenanceManager::getActivatedStatus()
+        {
+            JsonObject joGetParams;
+            JsonObject joGetResult;
+            std::string callsign = "org.rdk.AuthService.1";
+            std::string token;
+            uint8_t i = 0;
+            bool isAuthSerivcePluginActive = false;
+#if defined(BUILD_ENABLE_NOT_ACTIVATED)
+            std::string ret_status("not-activated");
+#else
+            std::string ret_status("activation-connect");
+#endif
+
+            /* check if plugin active */
+            isAuthSerivcePluginActive = Utils::isPluginActivated("org.rdk.AuthService");
+            if (!isAuthSerivcePluginActive){
+                LOGINFO("AuthService plugin is not activated.Retrying.. \n");
+                //if plugin is not activated we need to retry
+                do{
+                    isAuthSerivcePluginActive = Utils::isPluginActivated("org.rdk.AuthService");
+                    if ( !isAuthSerivcePluginActive ){
+                        sleep(10);
+                        i++;
+                        LOGINFO("AuthService retries [%d/4] \n",i);
+                    }
+                    else{
+                        break;
+                    }
+                }while( i < MAX_ACTIVATION_RETRIES );
+
+                if ( !isAuthSerivcePluginActive ){
+                    LOGINFO("AuthService plugin is Still not active");
+                    return ret_status;
+                }
+                else{
+                    LOGINFO("AuthService plugin is Now active");
+                }
+            }
+
+            Utils::SecurityToken::getSecurityToken(token);
+
+            Core::SystemInfo::SetEnvironment(_T("THUNDER_ACCESS"), _T(SERVER_DETAILS));
+            auto thunder_client = make_shared<WPEFramework::JSONRPC::LinkType<WPEFramework::Core::JSON::IElement> >(callsign.c_str(), "");
+            if (thunder_client != nullptr) {
+                uint32_t status = thunder_client->Invoke<JsonObject, JsonObject>(5000, "getActivationStatus", joGetParams, joGetResult);
+                LOGINFO("Invoke status : %d",status);
+                if (status > 0) {
+                    LOGINFO("%s call failed %d", callsign.c_str(), status);
+#if defined(BUILD_ENABLE_NOT_ACTIVATED)
```

Comment:
Pls remove these activation status defaults. These are internal to authService getActivationStatus API.

Reply:
removed.